hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -1511,6 +1629,71 @@ private void parsePB(BucketCacheProtos.BucketCacheEntry proto) throws IOExceptio
     verifyCapacityAndClasses(proto.getCacheCapacity(), proto.getIoClass(), proto.getMapClass());
   }
 
+  private void persistChunkedBackingMap(FileOutputStream fos) throws IOException {
+    fos.write(PB_MAGIC_V2);
+    int backingMapSize = backingMap.size();
+    long numChunks = backingMap.size() / persistenceChunkSize;
+    if (backingMap.size() % persistenceChunkSize != 0) {
+      numChunks += 1;
+    }
+
+    LOG.debug("persistToFile: before persisting backing map size: {}, "
+        + "fullycachedFiles size: {}, chunkSize: {}, numberofChunks: {}",
+      backingMap.size(), fullyCachedFiles.size(), persistenceChunkSize, numChunks);
+
+    fos.write(convertToBytes(persistenceChunkSize));
+    fos.write(convertToBytes(numChunks));
+
+    int blockCount = 0;
+    int chunkCount = 0;
+
+    BucketCacheProtos.BackingMap.Builder builder = BucketCacheProtos.BackingMap.newBuilder();
```

Comment:
Why are we creating proto builder here? Can't this go to BucketProtoUtils?

Reply:
ack!